apps/backend/db_patches/0140_CreateUpdateUserIdFunc.sql

```diff
@@ -0,0 +1,69 @@
+DO
+$DO$
+BEGIN
+	IF register_patch('CreateUpdateUserIdFunc.sql', 'Chi Kai Lam', 'Update user id for merging', '2024-02-09') THEN
+    BEGIN
+      
+      CREATE OR REPLACE FUNCTION UPDATE_USER_ID(INOUT P_OLD_USER_ID BIGINT, INOUT P_NEW_USER_ID BIGINT)
+      LANGUAGE PLPGSQL AS
+      $$
+      DECLARE
+        TOTAL_ROWS           INTEGER := 0;
+        OLD_USER_ID_ROWS     INTEGER := 0;
+        OLD_USERS_REC        USERS%ROWTYPE;
+      BEGIN
+          IF P_OLD_USER_ID = P_NEW_USER_ID THEN 
+              RETURN;
+          END IF; 
+          SELECT COUNT(*) FROM USERS WHERE USER_ID = P_OLD_USER_ID OR USER_ID = P_NEW_USER_ID INTO TOTAL_ROWS;
+          IF TOTAL_ROWS = 2 THEN
+              SELECT * FROM USERS WHERE USER_ID = P_OLD_USER_ID INTO OLD_USERS_REC;
+              --DEAL WITH FK TABLES, WE CAN SAFELY UPDATE THEM NOW SINCE THE NEW ROW IN USERS
+              --DELETE ANY DUPLICATES THAT HAVE BEEN CREATED AGAINST THE NEW USER ID
+              DELETE FROM PROPOSAL_USER 
+              WHERE USER_ID = P_NEW_USER_ID 
+              AND PROPOSAL_PK IN 
+              (
+                  SELECT PROPOSAL_PK FROM PROPOSAL_USER 
+                  WHERE USER_ID = P_OLD_USER_ID
+              ); 
+              --UPDATE THE OLD USER NUMBER TO THE NEW USER NUMBER
+              UPDATE PROPOSAL_USER SET USER_ID = P_NEW_USER_ID WHERE USER_ID = P_OLD_USER_ID;
+              UPDATE PROPOSALS SET PROPOSER_ID = P_NEW_USER_ID WHERE PROPOSER_ID = P_OLD_USER_ID;
```

Comment:
Both here and in the case where only the old account exists we're updating id's in the `PROPOSAL_USER` and `PROPOSALS` tables. When I look at the database schema in DBeaver it lists a lot of other tables that refer to the user_id which I think could also need updating:
![image](https://github.com/UserOfficeProject/user-office-core/assets/2938284/75a4a933-9411-4902-a5b4-e80d2f61f237)